src/rust/storage/seg/src/datapool/memory.rs

```diff
@@ -0,0 +1,45 @@
+// Copyright 2021 Twitter, Inc.
+// Licensed under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+//! A simple memory backed datapool which stores a contiguous slice of bytes
+//! heap-allocated in main memory.
+
+use crate::datapool::Datapool;
+
+/// A contiguous allocation of bytes in main memory
+pub struct Memory {
+    data: Box<[u8]>,
+}
+
+impl Memory {
+    /// Create a new `Memory` datapool with the specified size (in bytes)
+    pub fn create(size: usize, prefault: bool) -> Self {
+        let data = if prefault {
+            let mut data = Vec::with_capacity(0);
```

Comment:
How does this compare to something like ` vec![0; size as usize]`? Someone on SO claimed it's more efficient than `resize`

Reply:
This is related to what I mentioned re: forcing the pages to be faulted - `vec![0; size]` results in an allocation without the pages becoming resident. I might change this to be the vec macro + a read (which could probably be every page size bytes instead of every byte).

Eitherway, the pattern here should be changed slightly and I'll add a comment about why we're using this pattern and a TODO for looking at doing faster initialization. The startup time doesn't seem too terrible with prefault enabled.

Reply:
Added comment + TODO.

LMK if that's sufficient or if I should address this further in this PR.

Reply:
this looks good